packages/vinxi/lib/router-modes.js

```diff
@@ -354,11 +354,23 @@ const routerModes = {
 								);
 							});
 
-							const transformedHtml = await viteDevServer.transformIndexHtml(
+							let transformedHtml = await viteDevServer.transformIndexHtml(
 								getRequestURL(event).href,
-								text,
+								html,
 							);
 
+							const { JSDOM } = await import("jsdom");
+							const jsdom = new JSDOM(transformedHtml);
+							const nodes = jsdom.window.document.querySelectorAll("[src],[href]");
+							nodes.forEach(node => {
+								const attr = node.hasAttribute("href") ? "href" : "src";
+								const path = node.getAttribute(attr) || "/";
+								if (!isAbsolute(path)) {
+									node.setAttribute(attr, join(viteDevServer.config.base, path));
+								}
+							});
+							transformedHtml = jsdom.serialize();
+
```

Comment:
The same can be achived by temporarly generating a index.html and letting vite transform it.

```js
const indexPath = join(router.root, "index.html");
writeFileSync(indexPath, text);
let transformedHtml = await viteDevServer.transformIndexHtml(
  "/index.html",
  text,
  event.node.req.originalUrl
);
rmSync(indexPath);
```

Reply:
Writing to disk is not necessary. This seems to do the trick.
```js
await viteDevServer.transformIndexHtml(
  "/index.html",
  text,
  getRequestURL(event).href
);
```